Sources/Widgets/Core/WidgetState/index.d.ts

```diff
@@ -0,0 +1,82 @@
+import { ObjectChain } from "lodash";
+import { vtkObject } from "../../../interfaces";
+
+export interface vtkWidgetState extends vtkObject {
+  /**
+   * Set the active flag of the widget state instance
+   * 
+   * @param active The active flag
+   */
+  setActive(active: boolean): boolean;
+
+  /**
+   * Get the active flag of the widget state instance
+   */
+  getActive(): boolean;
+
+  /**
+   * Bind a state to one or more labels. If no label is provided, the default one will be used.
+   * 
+   * @param subState The state to bound.
+   * @param labels The labels to which the state should be bound.
+   */
+  bindState(subState: vtkWidgetState, labels?: string | string[]): void;
+
+  /**
+   * Unbind a specific state from the widget state instance
+   * 
+   * @param subState The state to be unbound.
+   */
+  unbindState(subState: vtkWidgetState): void;
+
+  /**
+   * Unbind all states from the widget state instance
+   */
+  unbindAll(): void;
+
+  /**
+   * Activate the widget state instance. Same as calling `vtkWidgetState.setActive(true)`
+   */
+  activate(): void;
+
+  /**
+   * Deactivate thie widget state instance and all its sub states, except the `excludingState` argument.
+   * 
+   * @param excludingState A sub-state instance that should not be deactivated.
```

Comment:
`@param {vtkWidgetState} [excludingState] ...`

Reply:
Seems redundant since we already have the typescript type-hint?
Also why putting square brackets around `excludingState` ?

Reply:
it's used by jsdoc https://jsdoc.app/tags-param.html#optional-parameters-and-default-values